policy-controller/k8s/index/src/http_route.rs

```diff
@@ -0,0 +1,136 @@
+                            PathMatch::Exact(value.to_owned())
+                        }
+                        k8s_gateway_api::HttpPathMatch::PathPrefix { value } => {
+                            PathMatch::Prefix(value.to_owned())
+                        }
+                        k8s_gateway_api::HttpPathMatch::RegularExpression { value } => {
+                            PathMatch::Regex(value.to_owned())
+                        }
+                    });
+
+                let headers = route_match
+                    .headers
+                    .iter()
+                    .flatten()
+                    .map(|header_match| match header_match {
+                        k8s_gateway_api::HttpHeaderMatch::Exact { name, value } => HeaderMatch {
+                            name: name.to_owned(),
+                            value: Value::Exact(value.to_owned()),
+                        },
+                        k8s_gateway_api::HttpHeaderMatch::RegularExpression { name, value } => {
+                            HeaderMatch {
+                                name: name.to_owned(),
+                                value: Value::Regex(value.to_owned()),
+                            }
+                        }
+                    })
+                    .collect();
+
+                let query_params = route_match
+                    .query_params
+                    .iter()
+                    .flatten()
+                    .map(|query_param| match query_param {
+                        k8s_gateway_api::HttpQueryParamMatch::Exact { name, value } => {
+                            QueryParamMatch {
+                                name: name.to_owned(),
+                                value: Value::Exact(value.to_owned()),
+                            }
+                        }
+                        k8s_gateway_api::HttpQueryParamMatch::RegularExpression { name, value } => {
+                            QueryParamMatch {
+                                name: name.to_owned(),
+                                value: Value::Regex(value.to_owned()),
+                            }
+                        }
+                    })
+                    .collect();
+
+                let method = route_match.method.as_ref().and_then(|method| {
+                    match method.to_lowercase().as_str() {
+                        "connect" => Some(HttpMethod::CONNECT),
+                        "get" => Some(HttpMethod::GET),
+                        "post" => Some(HttpMethod::POST),
+                        "put" => Some(HttpMethod::PUT),
+                        "delete" => Some(HttpMethod::DELETE),
+                        "patch" => Some(HttpMethod::PATCH),
+                        "head" => Some(HttpMethod::HEAD),
+                        "options" => Some(HttpMethod::OPTIONS),
+                        "trace" => Some(HttpMethod::TRACE),
+                        _ => None,
+                    }
+                });
```

Comment:
this probably ought to be [`http::Method`'s `FromStr` impl...](https://docs.rs/http/latest/http/method/struct.Method.html#impl-FromStr)